sauna-app/pages/[saunaId].tsx

```diff
@@ -1,19 +1,58 @@
 import styles from '../styles/Home.module.css'
-import { useRouter } from 'next/router'
+import {SingleSaunaProps, SingleSauna } from '../lib/typeDefinitions'
+import fetchSingleSauna from '../lib/load-single-sauna'
+import fetchData from '../lib/load-saunas'
+import Link from 'next/link';
 
-function DetailPage() {
-    const router = useRouter()
-    const saunaId = router.query.saunaId
 
+function DetailPage(props: SingleSauna) {
     return (
         <div className={styles.container}>
+          <button className='backButton'><Link href='/'>Back</Link></button>
           <main className={styles.main}>
             <h1 className={styles.title}>
-              Sauna detail page!
+              {props.name}
             </h1>
+            <div className='saunaDetailCard'>
+              <p>Address: {props.address}</p>
+              <p>Description: {props.description}</p>
+              <p>Saunas: {props.saunaType.map(s=> <span>{s} | </span>)}</p>
+              <p>Swimming: {props.swimming.map(s=> <span>{s} | </span>)}</p>
+              <p>Services: {props.service.map(s=> <span>{s} | </span>)}</p>
+              <p>Price: {props.price} €</p>
+              <a href={`${props.website}`} target='_blank' >Link to webpage of sauna</a>
```

Comment:
Opening a website to a new tab is actually a bad practice - especially from accessibility perspective, unless there is a really good reason for it. But I'd say that in this case, there is no such reason, so I'd suggest removing the `target="_blank"`.